fix(ivy): save queries at the correct indices

kara opened the pull request:
Previous to this change, we were storing view queries at the
wrong index. This is because we were passing a raw index to the
store() instruction instead of an adjusted index (i.e. an
index that does not include the HEADER_OFFSET). We had an
additional issue where TView.blueprint was not backfilled
when TView.data was backfilled, so new component instances
created from the blueprint would end up having a shorter LView.
Both of these problems together led to the Material demo app
failing with Ivy. This commit fixes those discrepancies.

AndrewKushnir commented:
[Presubmit](https://test.corp.google.com/ui#id=OCL:230649922:BASE:230649954:1548302217989:848fa498)

kara committed:
fix(ivy): save queries at the correct indices

Previous to this change, we were storing view queries at the
wrong index. This is because we were passing a raw index to the
store() instruction instead of an adjusted index (i.e. an
index that does not include the HEADER_OFFSET). We had an
additional issue where TView.blueprint was not backfilled
when TView.data was backfilled, so new component instances
created from the blueprint would end up having a shorter LView.
Both of these problems together led to the Material demo app
failing with Ivy. This commit fixes those discrepancies.

kara commented:
[Newer presubmit](http://test/OCL:230656692:BASE:230656705:1548306917634:cb2bde2a)

AndrewKushnir reviewed (approved):
LGTM 👍
